src/golioth_memfault.c

```diff
@@ -42,7 +42,9 @@ K_TIMER_DEFINE(mflt_upload_timer, mflt_upload_timer_expired, NULL);
 
 static void coredump_timer_expired(struct k_timer *timer)
 {
+    #if !defined(CONFIG_SOC_FAMILY_ESP32)
     memfault_arch_disable_configurable_faults();
```

Comment:
@noahp I was actually confused why/if this was necessary. I added this because I was having trouble getting some crashes to trigger coredumps - is this the right way to fix that or is there some kconfig option we can enable?

Reply:
yeah, it's a nuance of the xtensa/riscv chips, I haven't added support for the arch-specific API we use here, so it's just compiled out on those chips for now